### DISCUSSION TITLE
Input tuple does not match warning

### DISCUSSION THREAD

**Discussion by jjyotikataria** [nextflow-io/nextflow#2176] [Q&A]
Hi Team @delagoya @nicoulaj @fstrozzi @robsyme @lindenb 

Its a warning but I really want to know the reason behind this occurence so that I can be fixing it and my script runs smoothly without any error.

WARN: Input tuple does not match input set cardinality declared by process `scrnabatch_qc` -- offending value: [GEX, PSC_GEX, /MG/SHARED/ANALYSIS/DEMUX/test_sc/TEST/FINAL/CHECK/work/ec/0960eb4a00f47879f8d562998e7aca/PSC_GEXGEX.analysis.over, GEX, CWB_GEX, /MG/SHARED/ANALYSIS/DEMUX/test_sc/TEST/FINAL/CHECK/work/e1/2973557fe56f5cbcc8b9d320155e0b/CWB_GEXGEX.analysis.over]

Here is the process block for the given warning:

process scrnabatch_qc {

        publishDir "$results", mode: 'copy', overwrite: true

        input:
        set val(sampleGEX),val(sample_name), file("${sample_name}.analysis.over") from ANALYSIS_OVER_GEX.collect()

        output:
        file 'scbatchqc.txt' into qc_ch
        file 'report.html' into qc_ch1

        script:
        if (params.species == "Human")

        """
        PATH=/MG/SHARED/APPS/R-3.6.1/bin:/home/jing.w/tools/pandoc-2.7.3/bin:$PATH

        export PATH

        Rscript $sc_scripts/qc_summary.R $results/all_filtered_bc_matrix/ hsapiens

        cp report.html $results/scRNABatchQC.html

        echo "sc_batch_qc_done" > scbatchqc.txt
        """

**Comment by lindenb** [comment]
X.collect() returns a List

>  collect operator collects all the items emitted by a channel to a List and return the resulting object as a sole emission

so your input is more like:

    val alist from ANALYSIS_OVER_GEX.collect()

**Reply by jjyotikataria** [reply]
So, how to avoid this then? @lindenb
